### ISSUE TITLE
Azure Devops task "Deploy Azure Static Web App" failing since version 0.200.0

### ISSUE THREAD

**Issue by yann-combarnous** [Azure/static-web-apps#731]
**Describe the bug**
Since the task was updated to 0.200.0, our CD pipeline fails to deploy to Azure Static Web App. Last known working version for the task was 0.194.2 .

**To Reproduce**
Run "Deploy Azure Static Web App" task in Azure Devops

Yaml for ADO task:
```
#Your build pipeline references an undefined variable named ‘AppToken’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972

steps:
- task: AzureStaticWebApp@0
  displayName: 'Static Web App QA Deployment'
  inputs:
    workingDirectory: '$(System.DefaultWorkingDirectory)/<our_repo>'
    app_location: build
    skip_app_build: true
    verbose: false
    azure_static_web_apps_api_token: '$(AppToken)'
```

staticwebapp.config.json:
```
{
  "navigationFallback": {
    "rewrite": "/index.html",
    "exclude": [
      "/static/*"
    ]
  },
  "globalHeaders": {
    "Server": "",
    "Content-Security-Policy": "<our CSP here>",
    "Strict-Transport-Security": "max-age=63072000; includeSubdomains",
    "X-Xss-Protection": "1; mode=block",
    "X-Content-Type-Options": "nosniff",
    "X-Frame-Options": "sameorigin",
    "Referrer-Policy": "strict-origin-when-cross-origin",
    "Feature-Policy": "geolocation *"
  },
  "responseOverrides": {
    "400": {
      "statusCode": 404,
      "rewrite": "/503.html"
    },
    "401": {
      "statusCode": 404,
      "rewrite": "/503.html"
    },
    "403": {
      "statusCode": 404,
      "rewrite": "/503.html"
    },
    "404": {
      "statusCode": 404,
      "rewrite": "/503.html"
    }
  },
  "mimeTypes": {
    ".json": "text/json"
  }
}
```

**Expected behavior**
Deployment task works.

**Screenshots**
Logs from task
```
2022-02-17T08:59:37.4690857Z ##[section]Starting: Static Web App QA Deployment
2022-02-17T08:59:37.4715060Z ==============================================================================
2022-02-17T08:59:37.4715470Z Task         : Deploy Azure Static Web App
2022-02-17T08:59:37.4715797Z Description  : [PREVIEW] Build and deploy an Azure Static Web App
2022-02-17T08:59:37.4716224Z Version      : 0.200.0
2022-02-17T08:59:37.4716469Z Author       : Microsoft Corporation
2022-02-17T08:59:37.4716817Z Help         : https://aka.ms/swadocs
2022-02-17T08:59:37.4717143Z ==============================================================================
2022-02-17T08:59:37.6481498Z [command]/bin/bash /opt/vsts-2/_work/_tasks/AzureStaticWebApp_18aad896-e191-4720-88d6-8ced4806941a/0.200.0/launch-docker.sh
2022-02-17T08:59:37.7539260Z unknown flag: --pull
2022-02-17T08:59:37.7539870Z See 'docker run --help'.
2022-02-17T08:59:37.7693354Z ##[error]Error: The process '/bin/bash' failed with exit code 125
2022-02-17T08:59:37.7713721Z ##[section]Finishing: Static Web App QA Deployment
```

**Comment by CristianQS** [comment]
Hi @yann-combarnous ! 😀  The **Deploy Azure Static Web App Task** executes the script **launch-docker.sh** 
```
docker run \
    --env-file ./env.list \
     --pull=always \
    -v "$SWA_WORKING_DIR:$SWA_WORKSPACE_DIR" \
    "$SWA_DEPLOYMENT_CLIENT" \
    ./bin/staticsites/StaticSitesClient run

```
Since **version 20.10.0, Docker** add the pull flag to **docker run** command. You can see [here](https://docs.docker.com/engine/release-notes/#client-7).

Probably you are running a docker version that is lower than version 20.10.0. To fix this, you can **update docker from 20.10.0**  or you can **modify the script launch-docker.sh**  and remove the  `--pull` flag. It will be like this

```
docker run \
    --env-file ./env.list \
    -v "$SWA_WORKING_DIR:$SWA_WORKSPACE_DIR" \
    "$SWA_DEPLOYMENT_CLIENT" \
    ./bin/staticsites/StaticSitesClient run
```

**Comment by anthonychu** [comment]
Is it possible to update Docker client on your agent? We’ll look into updating a future version to use a separate pull command instead of a flag to be more compatible with older clients.
